diff --git a/units/systemd-reboot.service.in b/units/systemd-reboot.service.in
index 9565f2c..a6fbb23 100644
--- a/units/systemd-reboot.service.in
+++ b/units/systemd-reboot.service.in
@@ -16,3 +16,5 @@ After=shutdown.target umount.target final.target
 Type=oneshot
 ExecStart=@SYSTEMCTL@ --force reboot
 ExecStartPre=/bin/sh -c " if test -e /run/systemd/reboot-param ; then read REBOOTPARAM < /run/systemd/reboot-param ; fw_setenv bootargs_mode $REBOOTPARAM ; fi "
+ExecStartPre=/bin/systemctl stop systemd-journald.socket
+ExecStartPre=/bin/systemctl stop systemd-journald
