From 7d58908e7e47b46640f4bb650110801b1667719b Mon Sep 17 00:00:00 2001
From: Ferry Toth <ftoth@exalondelft.nl>
Date: Sun, 18 Mar 2018 17:04:23 +0100
Subject: [PATCH 1/1] 8250_mid: toggle IO7 on ttyS1 interrupt entry

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>
---
 drivers/tty/serial/8250/8250_mid.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/tty/serial/8250/8250_mid.c b/drivers/tty/serial/8250/8250_mid.c
index 8ec03863606e..0be5ddfe8051 100644
--- a/drivers/tty/serial/8250/8250_mid.c
+++ b/drivers/tty/serial/8250/8250_mid.c
@@ -7,6 +7,7 @@
  */
 
 #include <linux/bitops.h>
+#include <linux/gpio.h>
 #include <linux/module.h>
 #include <linux/pci.h>
 #include <linux/rational.h>
@@ -88,6 +89,12 @@ static int tng_handle_irq(struct uart_port *p)
 
 	chip = pci_get_drvdata(mid->dma_dev);
 
+        /* Toggle IO7 pin which corresponds to GP48*/
+	if(gpio_is_valid(48)) {
+		gpio_set_value(48, 1);
+		gpio_set_value(48, 0);
+	};
+
 	/* Rx DMA */
 	err = hsu_dma_get_status(chip, mid->dma_index * 2 + 1, &status);
 	if (err > 0) {
@@ -125,11 +132,17 @@ static int tng_setup(struct mid8250 *mid, struct uart_port *p)
 	mid->dma_dev = pci_get_slot(pdev->bus, PCI_DEVFN(5, 0));
 
 	p->handle_irq = tng_handle_irq;
+	/* allocate IO7 pin to toggle */
+	gpio_request(48, "ttyS0-2 INT");
+	gpio_direction_output(48, 0);
+
 	return 0;
 }
 
 static void tng_exit(struct mid8250 *mid)
 {
+	gpio_free(48);
+
 	pci_dev_put(mid->dma_dev);
 }
 
-- 
2.43.0

