--- a/gsupplicant/supplicant.c
+++ b/gsupplicant/supplicant.c
@@ -1963,8 +1963,11 @@
 	if (g_strcmp0(key, "Capabilities") == 0) {
 		supplicant_dbus_property_foreach(iter, interface_capability,
 								interface);
+#if 0
+		/* Disable temporarely p2p */
 		if (interface->mode_capa & G_SUPPLICANT_CAPABILITY_MODE_P2P)
 			interface->p2p_support = true;
+#endif
 	} else if (g_strcmp0(key, "State") == 0) {
 		const char *str = NULL;
 

